core/src/main/kotlin/org/jetbrains/research/testspark/core/generation/llm/prompt/PromptGenerator.kt

```diff
@@ -90,4 +90,32 @@ class PromptGenerator(
 
         return prompt
     }
+
+    /**
+     * Generates a prompt for a given line under test using CUT as a context.
+     *
+     * @param lineUnderTest The source code of the line to be tested.
+     * @param interestingClasses The list of `ClassRepresentation` objects related to the line under test.
+     * @param testSamplesCode The code snippet that serves as test samples.
+     * @return The generated prompt as `String`.
+     * @throws IllegalStateException If any of the required keywords are missing in the prompt template.
+     */
+    fun generatePromptForLine(
+        lineUnderTest: String,
+        interestingClasses: List<ClassRepresentation>,
+        testSamplesCode: String,
+    ): String {
+        val prompt = PromptBuilder(promptTemplates.linePrompt)
+            .insertLanguage(context.promptConfiguration.desiredLanguage)
+            .insertName(lineUnderTest.trim())
```

Comment:
Add information about context as well(as it was done for class generation)

For another type of generatePromptForLine, you can provide such information using psiHelper(as it was done for method generation)


This will resolve #377

Reply:
I implemented it in the commits [957bdb0](https://github.com/JetBrains-Research/TestSpark/pull/344/commits/957bdb06e56b56520366b7161d32a75b2e285a9d) and [441ae06](https://github.com/JetBrains-Research/TestSpark/pull/344/commits/441ae06ce627d105cbecc645bbf44cdb165432c0).

Adding only package was not enough, LLM wasn't able to instantiate a class containing a method that surrounds a selected line. Therefore, I insert a list of constructor signatures of the class into a prompt. It allows AI instantiate a class.